src/System.Private.CoreLib/src/System/Globalization/FormatProvider.FormatAndParse.Unix.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+using System.Globalization;
+using System.Diagnostics;
+
+namespace System.Globalization
+{
+    internal partial class FormatProvider
+    {
+        private partial class Number
+        {
+            // buffer size to hold digits (ECVT_MAX_COUNT_SIZE), decimal point, number sign, exponent, exponent symbol 'e', exponent sign and null.
+            private const int  ECVT_MAX_BUFFER_SIZE = 357;  
+            private const int  ECVT_MAX_COUNT_SIZE  = 348; // max digits count for floating numbers
+
+            internal static unsafe void _ecvt_s(byte* buffer, int sizeInBytes, double value, int count, int* dec, int* sign)
```

Comment:
does doing 2 stack allocation have significant cost compared to 1 allocation? I am asking because allocating one buffer and split it to use it for 2 purpose will not be the best of code clarity. 
I'll make the PAL on the level of DoibleToNumber